runtime/lua/vim/ui/img/screen.lua

```diff
@@ -0,0 +1,275 @@
+  -- some GUID; so, leverage CSI escape codes to query, which are supported
+  -- by modern Windows Terminal instances
+  if vim.env.WT_SESSION then
+    return M.__csi_size()
+  end
+
+  local ffi = require('ffi')
+
+  if not M.__def then
+    ffi.cdef([[
+      typedef unsigned long DWORD;
+      typedef unsigned short WORD;
+      typedef int BOOL;
+      typedef void* HANDLE;
+      typedef short SHORT;
+
+      typedef struct _COORD {
+        SHORT X;
+        SHORT Y;
+      } COORD;
+
+      typedef struct _SMALL_RECT {
+        SHORT Left;
+        SHORT Top;
+        SHORT Right;
+        SHORT Bottom;
+      } SMALL_RECT;
+
+      typedef struct _CONSOLE_SCREEN_BUFFER_INFO {
+        COORD dwSize;
+        COORD dwCursorPosition;
+        WORD wAttributes;
+        SMALL_RECT srWindow;
+        COORD dwMaximumWindowSize;
+      } CONSOLE_SCREEN_BUFFER_INFO;
+
+      typedef struct _CONSOLE_FONT_INFO {
+        DWORD nFont;
+        COORD dwFontSize;
+      } CONSOLE_FONT_INFO;
+
+      HANDLE GetStdHandle(DWORD nStdHandle);
+      BOOL GetConsoleScreenBufferInfo(
+        HANDLE hConsoleOutput,
+        CONSOLE_SCREEN_BUFFER_INFO* lpConsoleScreenBufferInfo
+      );
+      BOOL GetCurrentConsoleFont(
+        HANDLE hConsoleOutput,
+        BOOL bMaximumWindow,
+        CONSOLE_FONT_INFO* lpConsoleFontInfo
+      );
+    ]])
+    M.__def = true
+  end
+
+  ---@type vim.ui.img.screen.Size|nil
+  local size
+
+  ---Retrieve the screen buffer info and font size to determine the cell width and height.
+  ---NOTE: This does not work on Windows Terminal! We will fall back to CSI escape codes.
+  ---@type boolean, string|nil
+  local ok, err = pcall(function()
+    -- Using -11 should retrieve STD_OUTPUT_HANDLE, which initially is the
+    -- active console screen buffer (CONOUT$)
+    ---@type ffi.cdata*
+    local hStdOut = ffi.C.GetStdHandle(-11)
+
+    -- If our handle is INVALID_HANDLE_VALUE (-1)
+    if hStdOut == ffi.cast('HANDLE', -1) then
+      error('failed to get STD_OUTPUT_HANDLE')
+    end
+
+    ---@type { srWindow: { Left:integer, Top:integer, Right:integer, Bottom:integer } }
+    local csbi = ffi.new('CONSOLE_SCREEN_BUFFER_INFO')
+    if ffi.C.GetConsoleScreenBufferInfo(hStdOut, csbi) == 0 then
+      error('failed to retrieve screen buffer info')
+    end
+
+    ---@type { nFont:integer, dwFontSize: { X:integer, Y:integer } }
+    local fontInfo = ffi.new('CONSOLE_FONT_INFO')
+    if ffi.C.GetCurrentConsoleFont(hStdOut, false, fontInfo) == 0 then
+      error('failed to retrieve current console font')
+    end
+
+    -- Use the visible window (srWindow) to figure out the rows and columns shown
+    local cols = csbi.srWindow.Right - csbi.srWindow.Left + 1
+    local rows = csbi.srWindow.Bottom - csbi.srWindow.Top + 1
+
+    -- Use our font size as an approximation of the cell size
+    local cell_width = fontInfo.dwFontSize.X
+    local cell_height = fontInfo.dwFontSize.Y
```

Comment:
These values are meaningless. On Windows Terminal and the Windows console the cell size is unrelated to the font - it always emulates a VT340 cell size of 10x20. Other terminals running on Windows will typically have a variable cell size, but the values reported here would not reflect the font that the terminal is actually using. The only way to determine the real cell size is with an escape sequence.

Reply:
Got it, thanks for the clarification on this one. I was hoping @gpanders might see the conversation since he was the one providing insight on neovim's `TermResponse` and general query API. From what I can see, neovim consumes any CSI responses from the terminal, which is why I haven't implemented `CSI 14 t` or `CSI 16 t`.